lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -1162,6 +1162,13 @@ void TypeChecker::validateGenericTypeSignature(GenericTypeDecl *typeDecl) {
 
   prepareGenericParamList(gp, typeDecl);
 
+  // For a protocol, compute the requirement signature first. It will be used
+  // by clients of the protocol.
+  if (auto proto = dyn_cast<ProtocolDecl>(typeDecl)) {
```

Comment:
This has to go here because it needs the preparation of the previous two lines (and has to happen before the next one), right?